components/hab/src/main.rs

```diff
@@ -706,6 +754,352 @@ fn sub_pkg_channels(ui: &mut UI, m: &ArgMatches) -> Result<()> {
+        (Some(_org), Some(username)) => {
+            let user_pair = BoxKeyPair::get_latest_pair_for(username, &cache)?;
+            let service_pair = BoxKeyPair::get_latest_pair_for(&service_group, &cache)?;
+            ui.status(
+                Status::Encrypting,
+                format!(
+                    "TOML as {} for {}",
+                    user_pair.name_with_rev(),
+                    service_pair.name_with_rev()
+                ),
+            )?;
+            set.set_cfg(user_pair.encrypt(&buf, Some(&service_pair))?);
+            set.set_is_encrypted(true);
+        }
+        _ => set.set_cfg(buf.to_vec()),
+    }
+    set.set_service_group(service_group.into());
+    set.set_version(value_t!(m, "VERSION_NUMBER", u64).unwrap());
+    ui.begin(format!(
+        "Setting new configuration version {} for {}",
+        set.get_version(),
+        set.get_service_group(),
+    ))?;
+    ui.status(
+        Status::Creating,
+        format!("service configuration"),
+    )?;
+    SrvClient::connect(&sup_addr, &secret_key)
+        .and_then(|conn| {
+            conn.call(validate).for_each(
+                |reply| match reply.message_id() {
+                    "NetOk" => Ok(()),
+                    "NetErr" => {
+                        let m = reply.parse::<protocol::net::NetErr>().unwrap();
+                        match m.get_code() {
+                            ErrCode::InvalidPayload => {
+                                ui.warn(m)?;
+                                Ok(())
+                            }
+                            _ => Err(SrvClientError::from(m)),
+                        }
+                    }
+                    _ => {
+                        Err(SrvClientError::from(
+                            io::Error::from(io::ErrorKind::UnexpectedEof),
+                        ))
+                    }
+                },
+            )
+        })
+        .wait()?;
+    ui.status(
+        Status::Applying,
+        format!("via peer {}", sup_addr),
+    )?;
+    // JW: We should not need to make two connections here. I need a way to return the
+    // SrvClient from a for_each iterator so we can chain upon a successful stream but I don't
+    // know if it's possible with this version of futures.
+    SrvClient::connect(&sup_addr, secret_key)
+        .and_then(|conn| {
+            conn.call(set).for_each(|reply| match reply.message_id() {
+                "NetOk" => Ok(()),
+                "NetErr" => {
+                    let m = reply.parse::<protocol::net::NetErr>().unwrap();
+                    Err(SrvClientError::from(m))
+                }
+                _ => {
+                    Err(SrvClientError::from(
+                        io::Error::from(io::ErrorKind::UnexpectedEof),
+                    ))
+                }
+            })
+        })
+        .wait()?;
+    ui.end("Applied configuration")?;
+    Ok(())
+}
+
+fn sub_svc_config(m: &ArgMatches) -> Result<()> {
```

Comment:
This just effectively pulls back the package's `default.toml`, as opposed to whatever is currently circulating in the ring, right? Wonder if it's worth changing the subcommand name to reflect that.

(Also plays into https://github.com/habitat-sh/habitat/issues/4827)

I also notice this function is called as `hab config show`, so I'd expect this to be `sub_config_show`, or something similar.

Reply:
Precisely. This is a pretty old command that you couldn't even access without calling hab-sup directly. In the long time ago, this command was actually created so you could ask a _package_ and not a _service_ what it's configurables were. I think this needs to move to the pkg command and then this command needs to be expanded upon - we should be showing the entire config, not just defaults, including each section and the merged result.